src/components/TodoHeader/TodoHeader.tsx

```diff
@@ -0,0 +1,85 @@
+import cn from 'classnames';
+import {
+  useEffect, useRef, useState,
+} from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  setErrorMessage: (m: string) => void,
+  isDisable: boolean,
+  addTodo: (t: string, st: (t: string) => void) => void,
+  todos: Todo[],
+  updateTodo: (t: Todo) => void,
+};
+
+export const TodoHeader: React.FC<Props> = ({
+  setErrorMessage,
+  isDisable,
+  addTodo,
+  todos,
+  updateTodo,
+}) => {
+  const [todoTitle, setTodoTitle] = useState('');
+  const todoField = useRef<HTMLInputElement>(null);
+  const isActiveToggleAll = todos.every(todo => todo.completed === true);
+
+  useEffect(() => {
+    if (todoField.current) {
+      todoField.current.focus();
+    }
+  }, [isDisable]);
+
+  const handleFormSubmit = (ev: React.FormEvent<HTMLFormElement>) => {
+    ev.preventDefault();
+
+    if (!todoTitle.trim()) {
+      setErrorMessage('Title should not be empty');
+
+      setTimeout(() => {
+        setErrorMessage('');
+      }, 3000);
+
```

Comment:
try to keep this timeout in one place only (it App component) for not repeating this code